.github/workflows/ci-unit-tests.yml

```diff
@@ -35,13 +35,11 @@ jobs:
     - name: Run unit tests
       run: make test-ci
 
-    - name: Upload coverage reports to Codecov
-      uses: codecov/codecov-action@v5
+    - name: Upload coverage to codecov
+      uses: ./.github/actions/upload-codecov
```

Comment:
I don't think we need a sub-action here, it's overkill. We won't have multiple coverage files generated for this lib the way it was done in /jaeger/, so just inline the retries